### PR TITLE
feat: add SDGs to impact cards

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1586
Sanity PR: https://github.com/regen-network/regen-sanity/pull/30

- add sdgs to ProjectImpactCard
- wire sdgs to ProjectImpactCard
- remove sdgs from ProjectTopSection
- enhance impact section responsive

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by flagrede** [comment]
@erikalogie you can check the SDGs on ImpactCard here: https://deploy-preview-1846--regen-registry.netlify.app/project/C01-001

**Comment by erikalogie** [comment]
@flagrede this looks pretty good, I'm just not sure exactly what we want to do with the Verra logos, if we should keep those or get rid of them. @S4mmyb wdyt about getting rid of the Verra logos on these cards?

Also, it would be nice if the SDGs were scaled down to be smaller on mobile so that 3 can fit together on one line. Can you make it in Sanity such that only a max of 3 can be added?

Can you make it such that the SDGs are all on the same horizontal axis regardless of the length of the text?  So it would look like this: https://monosnap.com/file/lo8FcOWDm5s1WnoebMlA0IwYU38vlK

**Comment by flagrede** [comment]
@erikalogie sure I can do that.

**Comment by flagrede** [comment]
@erikalogie I handled your feedback, let me know if everything is ok now.

**Comment by flagrede** [comment]
@erikalogie you can give it an other try.

File: web-registry/src/components/organisms/ProjectImpactSection/ProjectImpactSection.utils.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { Maybe, Sdg } from 'generated/sanity-graphql';
+
+type GetSdgsImagesParams = {
+  sdgs?: Maybe<Maybe<Sdg>[]>;
+};
+
+export const getSdgsImages = ({ sdgs }: GetSdgsImagesParams) => {
+  const sdgsImages: ImageType[] =
+    sdgs?.map(sdg => ({
+      src: String(sdg?.image?.image?.asset?.url ?? sdg?.image?.imageHref ?? ''),
```
Comment (blushi) [inline web-registry/src/components/organisms/ProjectImpactSection/ProjectImpactSection.utils.ts:12]: we could use `getSanityImgSrc` here

File: web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx [hunk 2]
```diff
@@ -69,7 +66,7 @@ function ProjectTopSection({
   const theme = useTheme();
   const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
 
-  const { creditClass, creditClassVersion, offsetGenerationMethod, sdgIris } =
+  const { creditClass, creditClassVersion, offsetGenerationMethod } =
     parseOffChainProject(offChainProject);
```
Comment (blushi) [inline web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:70]: should we remove the part related to parsing sdg iris within `parseOffChainProject`?

**Review by blushi — APPROVED** [review]
utACK, just a few nits

**Review by wgwz — APPROVED** [review]
utACK

just to confirm my understanding, SDGs will no longer be tied to what a particular project wants displayed. instead, they are tied to the ecological impacts, within sanity. how are the ecological impacts tied to a project again? does that also happen in sanity?

**Comment by blushi** [comment]
> how are the ecological impacts tied to a project again? does that also happen in sanity?

no, from the postgres `credit_class_version.metadata` referencing the impact IRIs

**Review by haveanicedavid — APPROVED** [review]
utACK

File: web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx (hunk 2 above)
Comment (flagrede) [reply web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:70]: The same hook is still used to create impact data so we need to keep SDG iris parsing there. On the other hand, I removed the associated SDGs code in `ProjectTopSection` [here](https://github.com/regen-network/regen-web/pull/1846/files#diff-b8a47eb57a43900bc803c9720c608dcc8d0023fc15356196bd1ef98a176ae9d7L99-L116).

Comment (blushi) [reply web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:70]: I'm not sure to get it, where else is this used?

**Pull request merged**